### PR TITLE
browser improvement

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**

This PR correctly updates the browser address bar when the origin is changed

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/185

### CODE REVIEW

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -857,6 +857,12 @@ export const BrowserTab = (props) => {
 	 */
 	const onLoadStart = async ({ nativeEvent }) => {
 		const { hostname } = new URL(nativeEvent.url);
+
+		if (nativeEvent.url !== url.current) {
+			changeUrl({ ...nativeEvent }, 'end-promise');
+			changeAddressBar({ ...nativeEvent }, 'end-promise');
+		}
+
```
Comment (jpcloureiro) [inline app/components/Views/BrowserTab/index.js:865]: I see that both `changeUrl` and `changeAddressBar` are called later inside this `onLoadStart` function. `changeUrl` is called unconditionally but `changeAddressBar` is called only on Android devices.
Should we maintain that logic?

I also noticed that both `changeUrl` and `changeAddressBar` accept a `type` parameter but they are not using it inside their body.

Comment (blackdevelopa) [reply app/components/Views/BrowserTab/index.js]: You are right. The type isn't used so I removed it. For the former, I might have to revert on it.

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
```
Scenario: User taps on spoof link in browser
```
Given I have MetaMask mobile open on my device
And I am on the browser screen
And I have a tab open on unknown site
When I tap on spoof link to "https://opensea.io"
And I am redirected to "https://opensea.io"
And I tap back for the browser
Then spoof page is displayed
And browser address bar displays spoof page url
And "opensea.io" is not displayed
```
```

**Review by chrisleewilcox — APPROVED** [review]
LGTM!

**Pull request merged**